### ISSUE TITLE
Where did /opt/mssql/bin/sqlpackage go?

### ISSUE THREAD

**Issue by spinningarrow** [Microsoft/mssql-docker#135]
I just pulled the latest `microsoft/mssql-server-linux` image and it doesn't appear to have the sqlpackage binary any more. It used to be at /opt/mssql/bin. Was it moved somewhere else?

**Comment by twright-msft** [comment]
It has been removed temporarily.  We are going to be updating DacFx to work on top of dotnet core and then sqlpackage will run on top of the new DacFx.  Sqlpackage will also be moved to it's own package or the mssql-tools package.  For now you can use sqlpackage on a Windows machine.  Sorry for the temporary inconvenience.

**Comment by spinningarrow** [comment]
I don't have access to a Windows machine. This makes the mssql-docker image unusable for me. Is there a way I can go back to an older version that does have `sqlpackage` in it?

**Comment by twright-msft** [comment]
Yes, you can pull/run the mssql-server-linux:ctp2-1 image.

**Comment by twright-msft** [comment]
Here's a couple of other options until we get this available again. Depending on your scenario, these might work just as well.
1)	restore an existing .bak file into the new image using the sqlcmd CLI tool. More info here: https://docs.microsoft.com/en-us/sql/linux/sql-server-linux-setup-tools  
2)	generate a T-SQL script to create the database and insert data. Then execute the T-SQL script with sqlcmd. The script can be generated with the “Generate Script” wizard in SSMS on Windows or the new Python-based mssql-scripter CLI tool on any OS.

**Comment by sanagama** [comment]
@spinningarrow, can you share more details about your scenario? How are you using sqlpackage?

**Comment by pcalchi** [comment]
I'm also using sqlpackage and had to revert back to ctp2-1.  Personally, docker is a beautiful thing in that it allows you to downgrade as quickly/simply as you can.

As far as my particular scenario, I use it on my across my team to run our local development environments.  My solution uses a mixture mongo and sql and I use docker-compose to stand up all the required infrastructure locally.

The workflow is the following.  We use Visual Studio along with sql database projects that produce our dacpacs.  If someone on the team needs to make a change to the schema, it's as simple as make the change, build the dacpac, then docker-compose, build and up.  It's a really great solution.  

We added two additional developers to the team in the last week and I've never seen anything better in my 15+ years of software development.

**Comment by pcalchi** [comment]
@twright-msft you mentioned that sqlpackage is in progress of an update, do you happen to know the timeline?  The reason I ask is because I'm having some issues deploying a rather complicated schema.

**Comment by twright-msft** [comment]
Thanks for sharing your story @pcalchi.  Docker is beautiful.  I'd be curious to see your Dockerfiles, entrypoint scripts, or docker-compose files if you wouldnt mind sharing them via pastebin or something.

Yes, Sqlpackage is getting updated to run on top of the new cross-platform DacFX and to be packaged differently.  That doesnt necessarily mean that we will be changing things around how a DACPAC is deployed though.  If you want to share the details of that with us via StackOverflow or something we could take a look at it.

**Comment by sanagama** [comment]
@pcalchi, since you're using Visual Studio and Database Projects to produce the dacpac, is running sqlpackage on Windows to deploy the dacpac to SQL running in Docker an option for your scenario?

**Comment by apboyle** [comment]
@sanagama @twright-msft, I've also had to pin my Dockerfile to a previous CTP so that I can continue to use sqlpackage.
In my case I'm building up an image containing a copy of our database for use during integration testing.

The Dockerfile includes something like:

```
RUN <script that downloads dacpac artifacts from TeamCity to /scripts folder>

RUN /opt/mssql/bin/sqlserver.sh & sleep 20 \
&& /opt/mssql/bin/sqlpackage /Action:Publish /TargetServerName:localhost /TargetUser:SA /TargetPassword:$SA_PASSWORD /SourceFile:/scripts/dacpac/MyDB.dacpac /Profile:/scripts/dacpac/MyDB.publish.xml /p:BlockOnPossibleDataLoss=false \
&& sleep 20 \
&& pkill sqlservr && sleep 10
&& rm -rf /scripts
```

The resulting image is used in a docker-compose file along with our application code. We can then quickly and easily stand up a fresh database every time we run through our integration tests.

I hope this is a useful example for you :)

**Comment by spinningarrow** [comment]
Sorry for the awfully late reply everyone, and thanks for chiming in. Here's my update:

> Yes, you can pull/run the mssql-server-linux:ctp2-1 image.

@twright-msft This works perfectly for me at the moment. I was looking for something like this, but being new to Docker, I wasn't able to easily find a list of tags for an image. 🙈

> can you share more details about your scenario? How are you using sqlpackage?

@sanagama I was using it to load in a small test database from a bacpac file when the sqlserver container starts, basically in order to have a small database for local development/testing.

**Comment by twright-msft** [comment]
You probably already figured this out @spinningarrow, but for others, if you go to Docker Hub you can see a list of tags on the Tags tab.
https://hub.docker.com/r/microsoft/mssql-server-linux/tags/
Docker Store has the same thing as do other container catalogs like the Red Hat Container Catalog.  
If you dont specify a tag then the image tagged as :latest will be pulled.

**Comment by james-mcelwain** [comment]
Hi @twright-msft, is there a better place to follow progress on the "new cross-platform DacFX"? I'm not super familiar with the MS ecosystem. Can this be tracked in dotnet/core?

We are super excited to be able to use these build tools cross-platform!

**Comment by twright-msft** [comment]
We havent moved dacfx into the open source domain yet so it's hard to follow the progress.  We are working on it though...

**Comment by eturan** [comment]
Hi @twright-msft 

with the latest update (2017-GA and 2017-latest)  /opt/mssql/bin/sqlservr executable is gone again. Is it in the image or removed completely?

Best,
E.T.

**Comment by twright-msft** [comment]
Removed completely for now.  Will put it back in the next few months.

**Comment by twright-msft** [comment]
We are working on sqlpackage on Linux right now.  I've seen it working.  Maybe could provide you a preview drop of it.  Shoot me a mail at twright @ reallybigsoftwarecompanyname dot com.  How much more time do you have left on ctp2-1?

**Comment by jordanhenderson** [comment]
Thank you, will do. We have ~11 days left in the evaluation period.

**Comment by necrosisbb** [comment]
I am in a similar situation we have been deploying using sqlpackage (our whole pipeline builds dacpac files and deploys to docker image with to mssql-docker on linux before running tests after the full env is brought up with docker-compse). We have been doing this for most of the year and are also sweating on the expiry time of ctp2-1, so it would be great if a we could use a preview as well.

**Comment by twright-msft** [comment]
@necrosisbb - ok please contact me through email and I'll see what we can do

**Comment by xkrt** [comment]
@twright-msft why not to publish the sqlpackage as docker image to docker hub with some 'beta' version tag?

**Comment by borgdylan** [comment]
@twright-msft Will the new sqlpackage (once it hits preview/GA) be added to the tools packages? I would not like to have to install sql server on each dev machine. I prefer hosting dev DBs on one machine and only have the tools like sqlcmd installed on all machines.

**Comment by twright-msft** [comment]
Yes, it will be either in its own package or in the mssql-tools package.  We might put it in its own package because it is fairly large and it has a dependency on dotnet core.  Input on that decision is welcomed. :)

**Comment by weishenaustralia** [comment]
@twright-msft , ctp2.1 is going to expire in a few days. Before the new version (including sqlpackage) comes out in a few months, I think it's a good idea to release another evaluation version of ctp2.1 (essentially just to extend expiration), so we can use it without struggling to emailing you to get the preview package, building an image and fixing issues.

**Comment by nakahparis** [comment]
Any updates on SqlPackage availability in a container ?

**Comment by twright-msft** [comment]
Still in progress, but if you need an early copy, hit me up.

**Comment by borgdylan** [comment]
Are early test copies available outside a container, preferably as deb package or binary linux tarballs?

**Comment by twright-msft** [comment]
Yes, but just in a limited as needed basis for now.  Shoot me an email at twright - microsoft and I'll follow up with you via email with the details.

**Comment by aleks091** [comment]
@twright-msft are you guys planning to add the SqlPackage.exe into microsoft/mssql-server-windows-developer as well ? 

I am trying to generate my update script from the .dacpac in the container. 

$SqlPackagePath = 'C:\Program Files (x86)\Microsoft SQL Server\130\DAC\bin\SqlPackage.exe'
& $SqlPackagePath `
    /SourceFile:DB_TEST.dacpac `
    /Action:script /OutputPath:deploy.sql /Properties:CommentOutSetVarDeclarations=true `
    /TargetServerName:. /TargetDatabaseName:DB_TEST /TargetUser:sa /TargetPassword:$sa_password `
    /Variables:BasePath='C:\PostSnapshotScripts' /Variables:DeploymentEnvironment='Dev'


but I haven't been able to get SqlPackage.exe in my container yet.

The dacpac used to be in microsoft/mssql-server-windows-express:2016-sp1 but I can't use that version because my sql server schema will use Sql Server replication.

**Comment by twright-msft** [comment]
At some point yes, but its probably not imminent.  What have you tried to get sqlpackage.exe into your container?  Is there some issue with adding it via your own Dockerfile?

**Comment by aleks091** [comment]
@twright-msft Thanks for your quick response. 

I managed to get SqlPackage by installing Microsoft.Data.Tools.Msbuild. 
RUN & nuget install Microsoft.Data.Tools.Msbuild -Version 10.0.61026

Now I'm getting the following error when building my own SQLServer image with SQLServer and Replication features:
Sqlcmd: Error: Microsoft ODBC Driver 13 for SQL Server : Login failed for user 'User Manager\ContainerAdministrator'..

**Comment by twright-msft** [comment]
@aleks091 - Just to make sure - You are talking about a SQL Server in _Windows_ container now?

**Comment by aleks091** [comment]
@twright-msft 

For some reason I get the error: Sqlcmd: Error: Microsoft ODBC Driver 13 for SQL Server : Login failed for user 'User Manager\ContainerAdministrator'.. when using the latest FROM microsoft/windowsservercore

I finally got SqlServer with Replication enabled working on a windows container by using FROM microsoft/windowsservercore:10.0.14393.1480 when building my own dockerfile for SQLServer. 

Here is my dockerfile: https://gist.github.com/aleks091/831f1ed01cb7ca9f0d5dbdbe313bbab7/ 

Also I was getting a permissions issue when trying to execute SqlPackage.exe to create an update.sql file from a .dacpac  using a Microsoft.Data.Tools.Msbuild version that was not compatible with SqlServer 2017. This one was a bit tricky to figure out since the error log said that my log in failed. So I tried to use SqlServer 2017 with Microsoft.Data.Tools.Msbuild -Version 10.0.61710.120 to successfully create my update.sql file and create my DB. 

After some research and trial and error. I finally got my personal SQL server project running in a windows container using own docker file for SqlServer with replication enabled.

Thank you.

**Comment by seanfdnn** [comment]
Hi @twright-msft , any forecast on when this will be put back into /opt/mssql/bin? Or better yet, open sourced, or a .net core code sample made available of how to achieve?

**Comment by twright-msft** [comment]
Pretty soon it will be back in the bin.  Just going through testing now.  Look for it in the CU5 or CU6 probably.

**Comment by seanfdnn** [comment]
Where does the 200MB limit on BACPAC size in Blob storage come from? Blob supports into the terabytes? Is there a way around this? (or should I file another issue?)

**Comment by twright-msft** [comment]
It seems that this used to be the limit in block blobs but since May 31, 2016 it is now. 4.75 TB
https://docs.microsoft.com/en-us/rest/api/storageservices/understanding-block-blobs--append-blobs--and-page-blobs

Are you actually hitting a limit or just referring to the limit that is documented here?
https://docs.microsoft.com/en-us/azure/sql-database/sql-database-export

**Comment by seanfdnn** [comment]
Just referring to the limit that’s documented there – perhaps the documentation is out of date? That would be fantastic if that’s the case.

Sean

[ihs.com]<https://ihsmarkit.com/>

 Sean Dunn, CD, P.Eng., PMP
 Principal Software Engineer | Energy
 1331 Macleod Tr SE | Calgary, AB, Canada | T2G 0K3 |
 P: +1 403 815 3098
 sean.dunn@ihsmarkit.com<mailto:sean.dunn@ihsmarkit.com>

[Facebook]<http://facebook.com/Explore.IHS>[Twitter]<https://twitter.com/IHS>[LinkedIn]<http://linkedin.com/company/ihs>[YouTube]<http://youtube.com/ihschannel>[IHS Blogs]<http://blog.ihs.com/>[Google+]<https://plus.google.com/+IhsExplore>

**Comment by twright-msft** [comment]
Yeah, my guess is that it's the docs are out of date.  Are you going to give it a try?  Let us know how it works out.  If it seems to work OK, then we'll just update the docs to 4.75 TB.

**Comment by twright-msft** [comment]
Should be in the next couple of months.  If you need something more immediately than that we have a beta version I can provide you.  Shoot me a mail at twright a t m i c r o s o f t dot c o m.

**Comment by twright-msft** [comment]
Hi all!  We finally have a preview ready for sqlpackage that is built on dotnet core and is cross-platform!  Below are the links to download from. They are evergreen links, i.e. each day a new build is uploaded. This way any checked in bug fix is available the next day. Included in the .zip file is the preview EULA.
linux
https://go.microsoft.com/fwlink/?linkid=873926
osx
https://go.microsoft.com/fwlink/?linkid=873927
windows
https://go.microsoft.com/fwlink/?linkid=873928 
Release notes:
1.	The /p:CommandTimeout parameter is hardcoded to 120
2.	Build and deployment contributors are not supported
a.	Need to move to .NET Core 2.1 where System.ComponentModel.Composition.dll is supported
b.	Need to handle case-sensitive paths
3.	SQL CLR UDT types are not supported. 
a.	This includes SQL Server Types SqlGeography, SqlGeometry, & SqlHierarchyId
4.	Older .dacpac and .bacpac files that use Json serialization are not supported
5.	Referenced .dacpacs (e.g. master.dacpac) may not resolve due to issues with case-sensitive file systems

For lack of a better method, please provide any feedback you have here on this GitHub issue.

Thanks for giving it a try and letting us know how it goes!

**Comment by twright-msft** [comment]
Yes!  
https://docs.microsoft.com/en-us/sql/linux/sql-server-linux-setup-tools?view=sql-server-linux-2017

There is also a new open source command line tool that's super cool and way, way better than sqlcmd in my opinion - mssql-cli.
https://github.com/dbcli/mssql-cli

Microsoft engineers are the primary committers on that project.

**Comment by joshbooker** [comment]
thanks again!  Which of those has parity with sqlcmd.exe - meaning can directly execute output from:
sqlpackage.exe /a:Script

**Comment by joshbooker** [comment]
@brianjordan here is a test of doing extract, compare, update and scripting in sqlopsstudio command line using legacy sqlpackage & sqlcmd for windows plus cross-plat mssql-scripter:
https://github.com/joshbooker/sqlOS-Scripter-Extract-Compare-Update

I will try with these cross-plat tools soon.

Also please vote for this idea to add SSDT equivalent to sqlopsstudio:
https://github.com/Microsoft/sqlopsstudio/issues/389

**Comment by tnsholding** [comment]
@twright-msft trying to use the new cross-platform sqlpackage in a docker container. Cannot resolve reference to master.dacpac. I 'm copying master.dacpac together with my primary xxx.dacpac. 
How can I get sqlpackage to correctly resolve master.dacpac?

**Comment by apboyle** [comment]
@tnsholding, there's a known issue at the moment with case-sensitive file systems. I had to change the case to MASTER.DACPAC before sqlpackage could find it.

**Comment by twright-msft** [comment]
@tnsholding , @apboyle  - Thanks for reporting this issue.  We have a fix in the pipeline for this.

**Comment by moodmosaic** [comment]
Great news! Just wondering why there are different packages for each platform? Isn't the whole point of this to use just one package/set of assemblies across platforms?

Also, I've noticed there's no README in the (Windows) .ZIP file. As a consumer of Microsoft.SqlServer.Dac.dll, do I have to manually copy all these files in the program's output directory?

In general, it'd be easier if you provide a NuGet Package, perhaps in addition to those .ZIP files, so that the users can create local NuGet package feeds and have NuGet take care of any dependencies.

**Comment by twright-msft** [comment]
@moodmosaic - bit of a long story here... sqlpackage is dependent on a library called DacFX which in turn has historically been dependent on the full .NET framework.  Of course that just works on Windows.  In order to get sqlpackage to run on Linux we had two options - 1) run sqlpackage on the SQL Platform Abstraction Layer (SQL PAL) which also includes full .NET.  If you are not familiar with the SQL PAL, please read more about it [here](https://cloudblogs.microsoft.com/sqlserver/2016/12/16/sql-server-on-linux-how-introduction/) or 2) replatform DacFx on dotnet core.  Since #1 was relatively easy we did that at first and that was the original release of sqlpackage, but it's not the ideal user experience.  We also wanted to make sure that sqlpackage could run on macOS.  So, we decided that #2 was the better longer term solution and started working on replatforming DacFX to dotnet core.  We have other tools that depend on DacFx and some people develop solutions directly against DacFX so they will benefit from this replatforming too.  Now that the native sqlpackage with DacFX is available and runs on dotnet core the experience is much better and you can run it on macOS and Windows.  The packaging is slightly different for each of the different operating systems to provide a familiar installation experience on each platform.

The fwlinks that I posted above are just to some raw builds of sqlpackage not the official GA ones.  They are there as a preview of what is coming and especially for people trying to use sqlpackage in a Linux container to give us feedback while we are finalizing the release.

On Windows, you should be able to download/install the latest official from the [download center](https://www.microsoft.com/en-us/download/details.aspx?id=54106)



Good suggestion on the Nuget package.  I'm not sure if we have had many requests for that yet, but if there was enough demand for that we could certainly look into it.

**Comment by johnconger** [comment]
Is it possible using the fwlinks links above to deploy a dacpac file to MS SQL 2017 running *native* (not docker) on linux?  For example:

`dotnet sqlpackage.dll /Action:Publish /SourceFile:"MyDacpac.dacpac" /p:CreateNewDatabase=false /TargetConnectionString:"Data Source=localhost;User ID=sq;Password=pwd123;Database=MyDB;Pooling=False"
`

When i try to do this i get this error:
*** Could not deploy package.
Unable to connect to master or target server 'MyDB'. You must have a user with the same password in master or target server 'MyDB'.

From what i see online this points to a discrepancy between my MS SQL version and the version in the Dacpac.

Is this problem possible to solve or do the versions of DacFX prevent this?

**Comment by twright-msft** [comment]
Is this a typo issue?  You have ID=sq.  Maybe you meant ID=sa?

dotnet sqlpackage.dll /Action:Publish /SourceFile:"MyDacpac.dacpac" /p:CreateNewDatabase=false /TargetConnectionString:"Data Source=localhost;User ID=sq;Password=pwd123;Database=MyDB;Pooling=False"

**Comment by moodmosaic** [comment]
Thanks for the comments, @twright-msft and @pensivebrian.  I'm actually using the DacServices API directly so I'll be looking forward to a NuGet package release for .NET Standard.

Meanwhile, I *might* try to just consume what's inside the nightly builds―if so, I'll let you know how it goes.

**Comment by johnconger** [comment]
@twright-msft I'm embarrassed to say that the typo was my issue.  Thanks! and sorry for wasting your time.

**Comment by SamuelMarks** [comment]
Thanks @twright-msft https://github.com/Microsoft/mssql-docker/issues/135#issuecomment-389245587

Until an official build is released, I've built an image https://hub.docker.com/r/samuelmarks/mssql-server-fts-sqlpackage-linux:
```bash
docker run -d -e 'ACCEPT_EULA=Y' -e 'SA_PASSWORD=<YourStrong!Passw0rd>' -p 1433:1433 --name sqlfts0 mssql-server-fts-sqlpackage-linux
docker exec -it sqlfts0 /opt/mssql-tools/bin/sqlcmd -S localhost -U SA -P '<YourStrong!Passw0rd>' -Q 'CREATE DATABASE MyDb0'
docker cp ~/Downloads/foo.bacpac sqlfts0:/opt/downloads/foo.bacpac
docker exec -it sqlfts0 dotnet /opt/sqlpackage/sqlpackage.dll /tsn:localhost /tu:SA /tp:'<YourStrong!Passw0rd>' /A:Import /tdn:MyDb0 /sf:foo.bacpac
```

PS: If anyone has pointers on implementing Udt, I'm all ears… getting this error from my .bacpac sqlpackage import:
> Type Udt is not supported on this platform.

**Comment by SamuelMarks** [comment]
Thanks @twright-msft https://github.com/Microsoft/mssql-docker/issues/135#issuecomment-389245587

Until an official build is released, I've built an image https://hub.docker.com/r/samuelmarks/mssql-server-fts-sqlpackage-linux:
```bash
docker run -d -e 'ACCEPT_EULA=Y' -e 'SA_PASSWORD=<YourStrong!Passw0rd>' -p 1433:1433 --name sqlfts0 mssql-server-fts-sqlpackage-linux
docker exec -it sqlfts0 /opt/mssql-tools/bin/sqlcmd -S localhost -U SA -P '<YourStrong!Passw0rd>' -Q 'CREATE DATABASE MyDb0'
docker cp ~/Downloads/foo.bacpac sqlfts0:/opt/downloads/foo.bacpac
docker exec -it sqlfts0 dotnet /opt/sqlpackage/sqlpackage.dll /tsn:localhost /tu:SA /tp:'<YourStrong!Passw0rd>' /A:Import /tdn:MyDb0 /sf:foo.bacpac
```

BTW: If anyone has pointers on implementing Udt, I'm all ears… getting this error from my .bacpac sqlpackage import:
> Type Udt is not supported on this platform.

**Comment by pensivebrian** [comment]
We now have some proper [sqlpackage documentation](https://docs.microsoft.com/en-us/sql/tools/sqlpackage?view=sql-server-2017), along with [installation docs](https://docs.microsoft.com/en-us/sql/tools/sqlpackage-download?view=sql-server-2017). The install docs have link to the latest sqlpackage builds and call out known issue in the [release notes](https://docs.microsoft.com/en-us/sql/tools/sqlpackage-release-notes?view=sql-server-2017#sqlpackage-on-macos-and-linux-001-preview). Going forward, all now builds will be noted in the docs.

@SamuelMarks - There a two issues with UDTs at the moment for the cross platform sqlpackage build:
1) We're currently using .NET Core 2.1. We need to move to .NET Core 2.1, which has fixes in SqlClient for handling UDTs. I hope to have a new sqlpackage build out soon with this change.
2) SQL Server CLR UDTs (SqlGeometry, SqlGeography, and SqlHierarchyId) still won't be supported since we don't have a version of the Microsoft.SqlServer.Types.dll that targets .NET Core. I don't have any timelines for a fix. However after fixing 1), you be able to use your own custom CLR UDTs provided you compile a version for .NET Core.

**Comment by johnconger** [comment]
@twright-msft I've been using the above link with a lot of success, but I just tried to use `dotnet sqlpackage.dll ...` on a new linux machine and after a few moments i get a `Aborted (core dumped)` message.  This happens every time.  Any ideas?

[createdump.zip](https://github.com/Microsoft/mssql-docker/files/2315882/createdump.zip)

**Comment by twright-msft** [comment]
Now that sqlpackage is release for Linux and MacOS, I'm going to close out this issue.  Thanks to everybody for your patience as we got this figured out and released!

**Comment by edwardaskew** [comment]
So is sqlpackage ever actually coming back to the docker image or is it always going to have to be installed on top of it? Niether :latest not :2019-CTP3.0-ubuntu seem to have it right now.

**Comment by ormico** [comment]
@pensivebrian Did sqlpackage ever make it into apt-get on Ubuntu? I'm not seeing it but previous comment in thread indicated it was planned to go there.

**Comment by johnymachine** [comment]
> Now that sqlpackage is release for Linux and MacOS, I'm going to close out this issue. Thanks to everybody for your patience as we got this figured out and released!

Where is good place to submit issues for linux version of sqlpackage? This thread is as close as I coul find.

**Comment by RiverHeart** [comment]
So, it seems like sqlpackage is still not in the Microsoft APT repo nor is it shipping in MSSQL Docker. I know we can just install the Linux version via Dockerfile but why isn't it just available out of the gate?

**Comment by pensivebrian** [comment]
Looping in @dzsquared about sqlpackage availability on Microsoft APT.

**Comment by dzsquared** [comment]
For submitting issues on SqlPackage: https://github.com/microsoft/dacfx

To install Sqlpackage on linux, including in your own dockerfile:
(requires the image includes .NET SDK, min version .NET 6)
```
dotnet tool install -g microsoft.sqlpackage
```

https://learn.microsoft.com/sql/tools/sqlpackage/sqlpackage-download

**Comment by RiverHeart** [comment]
> For submitting issues on SqlPackage: https://github.com/microsoft/dacfx
> 
> To install Sqlpackage on linux, including in your own dockerfile: (requires the image includes .NET SDK, min version .NET 6)
> 
> ```
> dotnet tool install -g microsoft.sqlpackage
> ```
> 
> https://learn.microsoft.com/sql/tools/sqlpackage/sqlpackage-download
> 
> Having an independent Dockerfile for SqlPackage isn't currently on the roadmap, but do share more about your use case for this on the GitHub repo for DacFx.

I have a multistage Dockerfile where I build an SDK style Database Project on a dotnet 7 image then copy the DACPACs over to a MSSQL image where I've installed sqlpackage and spin up the database to install the DACPAC before the final image is baked. The MSSQL image doesn't have dotnet installed so I'm installing unzip and downloading the Linux sqlpackage zip package. Not sure if that's the best way to handle it but it's what I've come up with so far. Then the plan is to attach the container with the updated schema to an existing database.
